lib/bundler/plugin.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+module Bundler
+  module Plugin
+    autoload :Api,        "bundler/plugin/api"
+    autoload :Dsl,        "bundler/plugin/dsl"
+    autoload :Index,      "bundler/plugin/index"
+    autoload :Installer,  "bundler/plugin/installer"
+    autoload :SourceList, "bundler/plugin/source_list"
+
+    PLUGIN_FILE_NAME = "plugin.rb".freeze
+
+    @commands = {}
+
+    class << self
```

Comment:
`module_function` doesn't provide the ability to define private methods on the module. That leaves us with two options: 1. Give up on singleton pattern for Plugin, 2. Make the private methods public.

I am aware that [bbatsov/ruby-style-guide](https://github.com/bbatsov/ruby-style-guide#module-function) recomends it, but I don't find it reflected in [GitHub's](https://github.com/airbnb/ruby) or [Airbnb's](https://github.com/styleguide/ruby)